### PR TITLE
DAOS-15695 ci: golang update to 1.22.3

### PR DESCRIPTION
**Pull request by grom72**
Update Golang due to detected vulnerability in the previous versions (1.21.6)

### CODE REVIEW

**Review by JohnMalmberg — DISMISSED** [review]

File: go.spec [hunk 1]
```diff
@@ -111,6 +111,9 @@ fi
 %endif
 
 %changelog
+* Thu Jan 18 2024 Tomasz Gromadzki <tomasz.gromadzki@intel.com> - 1.22.3-1
```
Comment (brianjmurrell) [inline go.spec]: Need to update this datestamp.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: go.spec (hunk 1 above)
Comment (grom72) [reply go.spec]: Done

**Comment by brianjmurrell** [comment]
Hrm.  The [downstream testing](https://jenkins-1.wolf.hpdd.intel.com:9090/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-go-build-toolchain-PR-25-master/7/pipeline/182) is not using this PR's packages in it's RPM build setup.  I suspect that's because it's being set with the `CI_PR_REPOS` and not the `PR-repos:` commit pragma.  We are probably only checking and using the latter.

Probably/perhaps https://github.com/daos-stack/packaging/blob/1567c66dd52acdab63e50009115f2acde9212bef/Makefile_packaging.mk#L38 needs to check for and prefer the `CI_PR_REPOS` environment variable if it exists.  Could you try patching the `packaging/` subdir in this pr with:

```diff
--- a/Makefile_packaging.mk
+++ b/Makefile_packaging.mk
@@ -35,7 +35,11 @@ TEST_PACKAGES ?= ${NAME}
 # unfortunately we cannot always name the repo the same as the project
 REPO_NAME ?= $(NAME)
 
+ifneq ($(CI_PR_REPOS),)
+PR_REPOS                 ?= $(CI_PR_REPOS)
+else
 PR_REPOS                 ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos: *\(.*\)/\1/p')
+endif
 LEAP_15_PR_REPOS         ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-leap15: *\(.*\)/\1/p')
 EL_7_PR_REPOS            ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-el7: *\(.*\)/\1/p')
 EL_8_PR_REPOS            ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-el8: *\(.*\)/\1/p')
```

on the hope that `CI_PR_REPOS` is exposed to the environment that `make` is running in?

If that doesn't work we might have to dig a bit deeper and see if that environment variable is set in the environment this script is running in: https://github.com/daos-stack/pipeline-lib/blob/31dfdfe7ae64715e2da0fadffaa0a8c3a4be3378/vars/buildRpm.groovy#L73-L74.  But let's just start with the patch above.  You may want to create a new PR based on this PR to do that testing in so as not to muck up this PR.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
LGTM, but I'm just going to -1 until we work out the downstream testing issue.

**Comment by brianjmurrell** [comment]
@grom72 My apologies.  I wasn't considering where `CI_PR_REPOS` needs to be evaluated.  That needs to happen in the daos repo, not this one.

Can you create a daos PR and apply the below patch to it in addition to it being applied here?

> ```diff
> --- a/Makefile_packaging.mk
> +++ b/Makefile_packaging.mk
> @@ -35,7 +35,11 @@ TEST_PACKAGES ?= ${NAME}
>  # unfortunately we cannot always name the repo the same as the project
>  REPO_NAME ?= $(NAME)
>  
> +ifneq ($(CI_PR_REPOS),)
> +PR_REPOS                 ?= $(CI_PR_REPOS)
> +else
>  PR_REPOS                 ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos: *\(.*\)/\1/p')
> +endif
>  LEAP_15_PR_REPOS         ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-leap15: *\(.*\)/\1/p')
>  EL_7_PR_REPOS            ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-el7: *\(.*\)/\1/p')
>  EL_8_PR_REPOS            ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-el8: *\(.*\)/\1/p')
> ```

Once you have that daos PR, you can add a new (empty, even) commit to this PR with the following in the commit message:

```
Test-master-branch: <daos-PR's/branch>
```

Note that `<daos-PR's/branch>` is the branch name (i.e. `grom72/use-CI_PR_REPOS`), not a PR-####.  You can push the updated commit to this PR as soon as you push the daos PR.  You don't need to wait for that PR to complete (or even be in) CI.

**Comment by grom72** [comment]
DAOS pipeline adjusted in https://github.com/daos-stack/daos/pull/14387

**Review by JohnMalmberg — APPROVED** [review]

**Review by ryon-jensen — APPROVED** [review]

**Comment by ryon-jensen** [comment]
[Build 23](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-go-build-toolchain-PR-25-release-2.6/23/pipeline/) had a valgrind error in NLT. @grom72 created a fix (suppression file). Then [build 26](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-go-build-toolchain-PR-25-release-2.6/26/pipeline) with the fix applied passed NLT but had a failure in HW tests. These failures look unrelated to a go package update. I think this should be okay to land now.

**Comment by grom72** [comment]
All HW tests passed in [Build 23](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-go-build-toolchain-PR-25-release-2.6/23/pipeline) (No changes between build 23 and 26)

**Pull request merged**